docs/grammar/Solidity.g4

```diff
@@ -189,23 +190,43 @@ locals[
 	(Semicolon | body=block);
 
 /**
- * Definitions of the special fallback and receive functions.
+ * Definition of the special fallback function.
  */
-fallbackReceiveFunctionDefinition
+fallbackFunctionDefinition
 locals[
-	boolean visibilitySet = false,
 	boolean mutabilitySet = false,
 	boolean virtualSet = false,
-	boolean overrideSpecifierSet = false
+	boolean overrideSpecifierSet = false,
+	boolean hasParameters = false
 ]
 :
-	kind=(Fallback | Receive) LParen RParen
+	kind=Fallback LParen (parameterList { $hasParameters = true; } )? RParen
 	(
-		{!$visibilitySet}? visibility {$visibilitySet = true;}
+		{!$visibilitySet}? External {$visibilitySet = true;}
 		| {!$mutabilitySet}? stateMutability {$mutabilitySet = true;}
 		| modifierInvocation
 		| {!$virtualSet}? Virtual {$virtualSet = true;}
 		| {!$overrideSpecifierSet}? overrideSpecifier {$overrideSpecifierSet = true;}
+	)*
+	( {$hasParameters}? Returns LParen returnParameters=parameterList RParen | {!$hasParameters}? )
```

Comment:
Does this mean that if a fallback function has input parameters, then it must have return parameters as well?

Reply:
Never mind. I should've read the docs first :)